### PR TITLE
feat: [CDS-74554]: k8s & helm rollback changes: Get Node information via API from CD Deployment Step for CV

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50961)

### CODE REVIEW

**Comment by yogesh-chauhan** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java` line 537 at r1](https://reviewable.io/reviews/harness/harness-core/50961#-NaBv3TdEbtQ-L4winag:-NaBv3TdEbtQ-L4winah:b7jz9mw) ([raw file](https://github.com/harness/harness-core/blob/720c48feecfa4e334d167ecb34de669dcad9a65b/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java#L537)):*
> ```Java
>   }
> 
>   @VisibleForTesting
> ```

do we need this annotation ?

**Comment by yogesh-chauhan** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java` line 539 at r1](https://reviewable.io/reviews/harness/harness-core/50961#-NaBv85REKYfPEL9dkRQ:-NaBv85REKYfPEL9dkRR:bnsp8e0) ([raw file](https://github.com/harness/harness-core/blob/720c48feecfa4e334d167ecb34de669dcad9a65b/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java#L539)):*
> ```Java
>   @VisibleForTesting
>   public List<K8sPod> tagNewPods(List<K8sPod> newPods, List<K8sPod> existingPods) {
>     Set<String> existingPodNames = existingPods.stream().map(K8sPod::getName).collect(Collectors.toSet());
> ```

do we need null check here ?
does this method need to be public ?

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 6 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50961)* status: 7 file(s) to review. 0 of 2 approvals obtained. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/K8sRollingDeployRollbackResponse.java` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/50961#-NaBvg_l3Ed8cGXJLyuy:-NaBvg_l3Ed8cGXJLyuz:b-ws7lzf) ([raw file](https://github.com/harness/harness-core/blob/720c48feecfa4e334d167ecb34de669dcad9a65b/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/K8sRollingDeployRollbackResponse.java#L35)):*
> ```Java
>   public List<K8sPod> getTotalK8sPodList() {
>     return k8sPodList;
>   }
> ```

do we need explicit getters here ?

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
> _[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java` line 537 at r1](https://reviewable.io/reviews/harness/harness-core/50961#-NaBv3TdEbtQ-L4winag:-NaBv3TdEbtQ-L4winah:b7jz9mw) ([raw file](https://github.com/harness/harness-core/blob/720c48feecfa4e334d167ecb34de669dcad9a65b/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java#L537)):_
> 
> > ```java
> >   }
> > 
> >   @VisibleForTesting
> > ```
> 
> do we need this annotation ?

removed

**Comment by namantalaycha10** [comment]
> 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/K8sRollingDeployRollbackResponse.java

Actually I used the different get method name so getter will not work here

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 7 files at r1, 50 of 51 files at r2, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50961)* status: 58 file(s) to review. 1 of 2 approvals obtained.

**Comment by namantalaycha10** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java [hunk 1]
```diff
@@ -533,6 +535,42 @@ public List<K8sPod> getPods(
     return k8sPods;
   }
 
+  public List<K8sPod> tagNewPods(List<K8sPod> newPods, List<K8sPod> existingPods) {
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java:538]: Are these methods different than the one used in other handlers?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java (hunk 1 above)
Comment (namantalaycha10) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java]: yaah right, made the update. Now using the comman method

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50961)* status: 59 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by TarunUba1 — APPROVED** [review]
add the description of PR and Testing you did around it otherwise everything looks goodd

**Review by abosii-harness — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Pull request merged**